Mathlib/LinearAlgebra/Matrix/Determinant/TotallyUnimodular.lean

```diff
@@ -0,0 +1,95 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Vladimir Kolmogorov, Ivan Sergeev
+-/
+import Mathlib.LinearAlgebra.Matrix.Determinant.Basic
+
+/-!
+# Totally unimodular matrices
+
+This file defines totally unimodular matrices and provides basic API for them.
+
+## Main definitions
+
+ - `Matrix.IsTotallyUnimodular`: a matrix is totally unimodular iff every square submatrix
+    (not necessarily contiguous) has determinant `0` or `1` or `-1`.
+
+## Main results
+
+ - `Matrix.isTotallyUnimodular_iff`: a matrix is totally unimodular iff every square submatrix
+    (possibly with repeated rows and/or repeated columns) has determinant `0` or `1` or `-1`.
+ - `Matrix.IsTotallyUnimodular.apply`: entry in a totally unimodular matrix is `0` or `1` or `-1`.
+
+-/
+
+open scoped Matrix
+
+variable {m n R : Type*} [CommRing R]
+
+/-- Is the matrix `A` totally unimodular? -/
+def Matrix.IsTotallyUnimodular (A : Matrix m n R) : Prop :=
```

Comment:
All the names in the file start with `Matrix`. Could you please wrap it in the `Matrix` namespace?
And maybe also add
```
variable (A : Matrix m n R)
```

Reply:
Yes, I will wrap it into a namespace. However, I will not write `variable (A : Matrix m n R)` as it would be an obfuscation of the code.